test/unit/utils.authentication.unit.ts

```diff
@@ -84,14 +85,19 @@ describe('authentication utilities', () => {
   });
 
   async function createUser() {
-    user.password = await hashPassword(user.password, 4);
+    bcrypt_hasher = new BcryptHasher(4);
+    user.password = await bcrypt_hasher.hashPassword(user.password);
```

Comment:
Similarly here. I would expect that the logic of password hashing should be implemented in a shared piece of code that's called both by runtime parts (repositories/controllers/services) and by tests.

Maybe I am missing something?

Reply:
@bajtos See https://github.com/strongloop/loopback4-example-shopping/pull/51#discussion_r258152498 ^